redshift/resource_redshift_datashare.go

```diff
@@ -208,6 +248,35 @@ func resourceRedshiftDatashareAddAllTables(tx *sql.Tx, shareName string, schemaN
 	return err
 }
 
+func resourceRedshiftDatashareAddTablesInSchema(tx *sql.Tx, shareName string, schemas string, tables string) error {
+	query := fmt.Sprintf("ALTER DATASHARE %s ADD SCHEMA %s", pq.QuoteIdentifier(shareName), pq.QuoteIdentifier(schemas))
+	log.Printf("[DEBUG] %s\n", query)
+	_, err := tx.Exec(query)
+	if err != nil {
+		// if the schema is already in the datashare we get a "duplicate schema" error code. This is fine.
+		if pqErr, ok := err.(*pq.Error); ok {
+			if string(pqErr.Code) == pqErrorCodeDuplicateSchema {
+				log.Printf("[WARN] Schema %s already exists in datashare %s\n", schemas, shareName)
+			} else {
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+	query = fmt.Sprintf("ALTER DATASHARE %s ADD TABLE %s", pq.QuoteIdentifier(shareName), pq.QuoteIdentifier(tables))
+	log.Printf("[DEBUG] %s\n", query)
+	_, err = tx.Exec(query)
+	return err
+}
+
+func resourceRedshiftDatashareRemoveTablesInSchema(tx *sql.Tx, shareName string, tables string) error {
+	query := fmt.Sprintf("ALTER DATASHARE %s REMOVE TABLE %s", pq.QuoteIdentifier(shareName), pq.QuoteIdentifier(tables))
```

Comment:
Have you tested that it will actually work for more than one table in the list? To my knowledge, this will produce an SQL like `ALTER DATASHARE "xyz" REMOVE TABLE "schema.table1,schema.table2"` which will try to remove the table named `"schema.table1,schema.table2"` as one name rather than 2 separate tables (due to double quotes). The same would apply to `ADD TABLE` query.


I'm asking, because I've faced a similar issue with removing users from groups and had to quote each name individually before joining. See https://github.com/brainly/terraform-provider-redshift/blob/master/redshift/resource_redshift_group.go#L228-L232